packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -739,7 +740,11 @@ export function resetComponentRoot(vm: VM) {
     for (let i = 0, len = children.length; i < len; i++) {
         const child = children[i];
 
-        if (!isNull(child) && !isUndefined(child.elm)) {
+        // VFragments are special; their .elm property does not point to the root element since they have no root,
+        // so we have to clean them up differently.
+        if (!isNull(child) && isVFragment(child)) {
+            removeFragmentChildren(child, vm);
```

Comment:
I debated just converting this entire function to traverse using a stack so we wouldn't need a separate function, but this felt like a cleaner way to isolate the impact of VFragments on non-vfragment use cases.